internal/app/backend/backend_service.go

```diff
@@ -94,28 +95,36 @@ func (s *backendService) FetchMatches(ctx context.Context, req *pb.FetchMatchesR
 	resultChan := make(chan mmfResult, len(req.GetProfile()))
 
 	var syncID string
+	var err error
 	if s.synchronizerEnabled() {
-		resp, err := s.synchronizer.Register(ctx, &internalpb.RegisterRequest{})
+		var resp *internalpb.RegisterResponse
+		resp, err = s.synchronizer.Register(ctx, &internalpb.RegisterRequest{})
 		if err != nil {
 			return nil, err
 		}
 
 		syncID = resp.GetId()
 	}
 
-	err := doFetchMatchesInChannel(ctx, s.cfg, s.mmfClients, req, resultChan)
+	err = doFetchMatchesReceiveMmfResult(ctx, s.cfg, s.mmfClients, req, resultChan)
```

Comment:
Currently you discard the proposals that already have spoken for tickets - which leads to work done by MMFs already being discarded. Effectively, till tickets are assigned, MMFs may not be able to make forward progress (they may keep picking the tickets that OM ignores - and OM keeps dropping the results).

The approach we can take (similar to what was in prior OM) is to not return the ignored tickets in the FilterTickets response. Thus the ignored tickets simply act as one more filter for FilterTickets method. 

In the Backend, the only change is that we add the tickets to a ignored ticket list (like you already have). No other proposal check. The other change happens in FilterTickets in the guts of statestore - not exposed through any other API.